applications/ShapeOptimizationApplication/python_scripts/algorithms/algorithm_penalized_projection.py

```diff
@@ -192,6 +193,8 @@ def __isConstraintActive(self, constraintValue):
 
     # --------------------------------------------------------------------------
     def __logCurrentOptimizationStep(self):
+        if self.data_logger.SensitivityHeatmapLogging():
+            ComputeSensitivityHeatmap(self.design_surface, self.objectives, self.constraints, self.optimization_iteration, self.mapper)
```

Comment:
this seems like a hack/intrusive
can you refactor so that you don't need to copy paste this everywhere?